test/extensions/common/aws/metadata_fetcher_test.cc

```diff
@@ -0,0 +1,301 @@
+  MockUpstream mock_result(mock_factory_ctx_.cluster_manager_, &request);
+  MockMetadataReceiver receiver;
+  EXPECT_CALL(request, cancel());
+  EXPECT_CALL(receiver, onMetadataSuccess(testing::_)).Times(0);
+  EXPECT_CALL(receiver, onMetadataError(testing::_)).Times(0);
+
+  // Act
+  fetcher_->fetch(message, parent_span_, receiver);
+  // Proper cancel
+  fetcher_->cancel();
+  // Re-entrant cancel
+  fetcher_->cancel();
+}
+
+TEST_F(MetadataFetcherTest, TestSpanPassedDown) {
+  // Setup
+  setupFetcher();
+  Http::RequestMessageImpl message;
+  MockUpstream mock_result(mock_factory_ctx_.cluster_manager_, "200", "not_empty");
+  MockMetadataReceiver receiver;
+
+  // Expectations for span
+  EXPECT_CALL(mock_factory_ctx_.cluster_manager_.thread_local_cluster_.async_client_,
+              send_(_, _, _))
+      .WillOnce(Invoke(
+          [this](Http::RequestMessagePtr&, Http::AsyncClient::Callbacks&,
+                 const Http::AsyncClient::RequestOptions& options) -> Http::AsyncClient::Request* {
+            EXPECT_TRUE(options.parent_span_ == &this->parent_span_);
+            EXPECT_TRUE(options.child_span_name_ == "AWS Metadata Fetch");
+            return nullptr;
+          }));
+
+  // Act
+  fetcher_->fetch(message, parent_span_, receiver);
+}
+
+TEST_F(MetadataFetcherTest, TestDefaultRetryPolicy) {
+  // Setup
+  setupFetcher();
+  Http::RequestMessageImpl message;
+  MockUpstream mock_result(mock_factory_ctx_.cluster_manager_, "200", "not_empty");
+  MockMetadataReceiver receiver;
+
+  EXPECT_CALL(mock_factory_ctx_.cluster_manager_.thread_local_cluster_.async_client_,
+              send_(_, _, _))
+      .WillOnce(Invoke(
+          [](Http::RequestMessagePtr&, Http::AsyncClient::Callbacks&,
+             const Http::AsyncClient::RequestOptions& options) -> Http::AsyncClient::Request* {
+            EXPECT_TRUE(options.retry_policy.has_value());
+            EXPECT_TRUE(options.buffer_body_for_retry);
+            EXPECT_TRUE(options.retry_policy.value().has_num_retries());
+            EXPECT_EQ(PROTOBUF_GET_WRAPPED_REQUIRED(options.retry_policy.value(), num_retries), 3);
+
+            EXPECT_TRUE(options.retry_policy.value().has_per_try_timeout());
+            EXPECT_EQ(PROTOBUF_GET_MS_REQUIRED(options.retry_policy.value(), per_try_timeout),
+                      5000);
+
+            EXPECT_TRUE(options.retry_policy.value().has_per_try_idle_timeout());
+            EXPECT_EQ(PROTOBUF_GET_MS_REQUIRED(options.retry_policy.value(), per_try_idle_timeout),
+                      1000);
+
+            const std::string& retry_on = options.retry_policy.value().retry_on();
+            std::set<std::string> retry_on_modes = absl::StrSplit(retry_on, ',');
+
+            EXPECT_EQ(retry_on_modes.count("5xx"), 1);
+            EXPECT_EQ(retry_on_modes.count("gateway-error"), 1);
+            EXPECT_EQ(retry_on_modes.count("connect-failure"), 1);
+            EXPECT_EQ(retry_on_modes.count("reset"), 1);
```

Comment:
`EXPECT_THAT(retry_on_modes, IsSupersetOf({"5xx","gateway-error","connect-failure","reset"}));`
OR
`UnorderedElementsAre` is more precise if it's expected to be only those.